cli/src/cluster_query.rs

```diff
@@ -1786,6 +1793,47 @@ pub fn process_show_stakes(
             ]);
         }
     }
+
+    if let Some(withdraw_authority_pubkey) = withdraw_authority_pubkey {
+        match program_accounts_config.filters {
+            Some(filters) => {
+                program_accounts_config.filters = Some(
+                    [
+                        filters, // Filter by withdrawer
+                        vec![rpc_filter::RpcFilterType::Memcmp(rpc_filter::Memcmp {
+                            offset: 44,
+                            bytes: rpc_filter::MemcmpEncodedBytes::Base58(
+                                withdraw_authority_pubkey.to_string(),
+                            ),
+                            encoding: Some(rpc_filter::MemcmpEncoding::Binary),
+                        })],
+                    ]
+                    .concat(),
+                )
+            }
+            None => {
+                program_accounts_config.filters = Some(vec![
+                    // Filter by `StakeState::Stake(_, _)`
+                    rpc_filter::RpcFilterType::Memcmp(rpc_filter::Memcmp {
+                        offset: 0,
+                        bytes: rpc_filter::MemcmpEncodedBytes::Base58(
+                            bs58::encode([2, 0, 0, 0]).into_string(),
+                        ),
+                        encoding: Some(rpc_filter::MemcmpEncoding::Binary),
+                    }),
```

Comment:
You can get rid of this filter, since withdraw authority exists on `StakeState::Stake(_,_)` and `StakeState::Initialized(_)`